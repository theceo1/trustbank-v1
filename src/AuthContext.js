import React, { createContext, useContext, useState } from 'react';

const AuthContext = createContext();

export const AuthProvider = ({ children }) => {
  const [user, setUser] = useState(null);

  const login = async ({ email, password }) => {
    // Simulate an API call for login
    if (email === 'testing4@test.com' && password === 'password') {
      setUser({ email });
    } else {
      throw new Error('Invalid credentials');
    }
  };

  const logout = () => {
    setUser(null);
  };

  return (
    <AuthContext.Provider value={{ user, login, logout }}>
      {children}
    </AuthContext.Provider>
  );
};

export const useAuth = () => {
  return useContext(AuthContext);
};
