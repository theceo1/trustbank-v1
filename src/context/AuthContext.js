// src/context/AuthContext.js
import React, { createContext, useContext, useState, useEffect } from 'react';
import { useRouter } from 'next/router';
import { setCookie, parseCookies, destroyCookie } from 'nookies';
import jwt_decode from 'jwt-decode'; 
import axios from 'axios';

const AuthContext = createContext();

export const AuthProvider = ({ children }) => {
  const [user, setUser] = useState(null);
  const router = useRouter();

  useEffect(() => {
    const cookies = parseCookies();
    const token = cookies.token;

    console.log('Cookies:', cookies);
    console.log('Token:', token);

    if (token) {
      try {
        const decodedToken = jwt_decode(token); 
        console.log('Decoded Token:', decodedToken);
        setUser(decodedToken);
      } catch (error) {
        console.error('Failed to decode token:', error);
        destroyCookie(null, 'token');
        setUser(null);
      }
    }
  }, []);

  const signup = async ({ name, email, password }) => {
    try {
      const response = await axios.post('/api/signup', { name, email, password });
      console.log('Signup Response:', response);

      if (response.status === 200) {
        const { token, user } = response.data;
        setCookie(null, 'token', token, { path: '/' });
        setUser(user);
        router.push('/dashboard');
      }
    } catch (error) {
      console.error('Signup failed:', error);
    }
  };

  const signin = async ({ email, password }) => {
    try {
      const response = await axios.post('/api/signin', { email, password });
      console.log('Signin Response:', response);

      if (response.status === 200) {
        const { token, user } = response.data;
        setCookie(null, 'token', token, { path: '/' });
        setUser(user);
        router.push('/dashboard');
      }
    } catch (error) {
      console.error('Signin failed:', error);
    }
  };

  const signout = () => {
    destroyCookie(null, 'token');
    setUser(null);
    router.push('/');
  };

  return (
    <AuthContext.Provider value={{ user, signup, signin, signout }}>
      {children}
    </AuthContext.Provider>
  );
};

export const useAuth = () => useContext(AuthContext);
