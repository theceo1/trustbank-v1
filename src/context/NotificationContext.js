import React, { createContext, useContext, useState, useCallback } from 'react';
import { v4 as uuidv4 } from 'uuid';

const NotificationContext = createContext();

export const useNotification = () => useContext(NotificationContext);

export const NotificationProvider = ({ children }) => {
  const [notifications, setNotifications] = useState([]);

  const addNotification = useCallback((type, message) => {
    const id = uuidv4();
    setNotifications((prev) => [...prev, { id, type, message }]);
    setTimeout(() => {
      setNotifications((prev) => prev.filter((notification) => notification.id !== id));
    }, 5000);
  }, []);

  return (
    <NotificationContext.Provider value={{ notifications, addNotification }}>
      {children}
    </NotificationContext.Provider>
  );
};
