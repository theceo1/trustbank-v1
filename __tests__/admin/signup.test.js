import React from 'react';
import { render } from '@testing-library/react';
import SignUp from '@/pages/admin/signup';
import { AuthProvider } from '@/context/AuthContext';

const renderWithAuthProvider = (ui, { providerProps, ...renderOptions } = {}) => {
  return render(<AuthProvider {...providerProps}>{ui}</AuthProvider>, renderOptions);
};

describe('SignUp', () => {
  it('renders SignUp component', () => {
    const { asFragment } = renderWithAuthProvider(<SignUp />, {
      providerProps: { value: { user: null } },
    });
    expect(asFragment()).toMatchSnapshot();
  });
});
