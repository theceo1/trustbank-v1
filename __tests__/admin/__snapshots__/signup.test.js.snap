// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp renders SignUp component 1`] = `
<DocumentFragment>
  <div
    class="bg-white p-8 rounded-lg shadow-lg"
  >
    <div
      class="text-center mb-4"
    >
      <img
        alt="Logo"
        data-nimg="1"
        decoding="async"
        height="50"
        loading="lazy"
        src="/_next/image?url=%2Fimages%2Flogo.png&w=128&q=75"
        srcset="/_next/image?url=%2Fimages%2Flogo.png&w=64&q=75 1x, /_next/image?url=%2Fimages%2Flogo.png&w=128&q=75 2x"
        style="color: transparent;"
        width="50"
      />
    </div>
    <h2
      class="text-2xl font-bold mb-4 text-center"
    >
      Sign Up to Admin
    </h2>
    <form>
      <div
        class="mb-4"
      >
        <label
          for="name"
        >
          Name
        </label>
        <input
          class="w-full p-2 border rounded"
          id="name"
          placeholder="Your Name"
          required=""
          type="text"
        />
      </div>
      <div
        class="mb-4"
      >
        <label
          for="email"
        >
          Email
        </label>
        <input
          class="w-full p-2 border rounded"
          id="email"
          placeholder="Your Email"
          required=""
          type="email"
        />
      </div>
      <div
        class="mb-4"
      >
        <label
          for="password"
        >
          Password
        </label>
        <input
          class="w-full p-2 border rounded"
          id="password"
          placeholder="Your Password"
          required=""
          type="password"
        />
      </div>
      <button
        class="bg-teal-500 text-white p-2 rounded hover:bg-teal-600"
        type="submit"
      >
        Sign Up
      </button>
    </form>
    <div
      class="mt-4 text-center"
    >
      <p
        class="text-gray-700"
      >
        Already have an account?
        <a
          href="/admin/signin"
        />
        <a
          class="text-blue-500"
        >
           Sign in
        </a>
      </p>
    </div>
  </div>
</DocumentFragment>
`;
